cxx-qt-gen/src/parser/parameter.rs

```diff
@@ -0,0 +1,14 @@
+// SPDX-FileCopyrightText: 2022 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+// SPDX-FileContributor: Andrew Hayzen <andrew.hayzen@kdab.com>
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+use syn::{Ident, Type};
+
+/// Describes a Parameter for a QObject
```

Comment:
Do you intend to reuse this struct for more than parsing the signals? If so, I think this documentation could be improved. I think you mean parameters for QObject methods? If this will only be used for signals, then let's rename it to a more specific name, perhaps ParsedSignalParameters.

Reply:
It'll likely be used for parameters in signals, methods, and invokables in the future (like the existing equivalent in extract.rs)

Reply:
How about renaming it to ParsedFunctionParameter? And the documentation: `Describes a single parameter for a function`

Reply:
Potentially :thinking: guess it's obvious enough that signals are actually functions and not enums (like on the Rust side). And then extra info will go into these later like describing any conversion that needs to occur on the C++ side etc.

Reply:
I got confused reading this because a "parameter for a QObject" isn't really a thing, but a parameter for a QObject's method is.

Reply:
Fixed :-)